### ISSUE TITLE
When tool changes Mesh resource on a MeshInstance, it randomly crashes.

### ISSUE THREAD

**Issue by alimalkhalifa** [godotengine/godot#25575]
**Godot version:**
Godot 3.1 beta 3

**OS/device including version:**
Windows 10, NVidia Quadro M1000M

**Issue description:**
I have a tool script that manipulates the Mesh resource on a MeshInstance.  When mesh is set to null (or when mesh is replaced) it randomly crashes.  No logs of why that might be happening

![crashpic](https://user-images.githubusercontent.com/6815425/52173205-86b08500-2790-11e9-91c2-09615db7eff0.png)

Script:
```
tool
extends MeshInstance

export(bool) var make_mesh setget _add_mesh, _has_mesh

func _ready():
	pass

func _add_mesh(value):
	if not value or _has_mesh():
		mesh = null
		return
	var st = SurfaceTool.new()
	st.begin(Mesh.PRIMITIVE_TRIANGLES)
	st.add_vertex(Vector3(1,0,1))
	st.add_vertex(Vector3(0,0,1))
	st.add_vertex(Vector3(0,0,0))
	
	st.add_vertex(Vector3(0,0,0))
	st.add_vertex(Vector3(1,0,0))
	st.add_vertex(Vector3(1,0,1))
	
	st.index()
	mesh = st.commit()

func _has_mesh():
	if mesh != null:
		return true
	return false
```

**Minimal reproduction project:**
[ToolCrashTest.zip](https://github.com/godotengine/godot/files/2825218/ToolCrashTest.zip)

**Comment by qarmin** [comment]
I have this backtrace:
```
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x3ef20) [0x7f4b99878f20] (??:0)
[2] Set<Object*, Comparator<Object*>, DefaultAllocator>::_find(Object* const&) const (/home/rafal/Pulpit/godot/./core/set.h:228)
[3] Set<Object*, Comparator<Object*>, DefaultAllocator>::find(Object* const&) (/home/rafal/Pulpit/godot/./core/set.h:522)
[4] Set<Object*, Comparator<Object*>, DefaultAllocator>::erase(Object* const&) (/home/rafal/Pulpit/godot/./core/set.h:558)
[5] Object::remove_change_receptor(Object*) (/home/rafal/Pulpit/godot/core/object.cpp:973)
[6] EditorInspector::edit(Object*) (/home/rafal/Pulpit/godot/editor/editor_inspector.cpp:1796)
[7] EditorInspector::_notification(int) (/home/rafal/Pulpit/godot/editor/editor_inspector.cpp:2144)
[8] EditorInspector::_notificationv(int, bool) (/home/rafal/Pulpit/godot/editor/editor_inspector.h:251 (discriminator 14))
[9] Object::notification(int, bool) (/home/rafal/Pulpit/godot/core/object.cpp:957)
[10] Node::_propagate_exit_tree() (/home/rafal/Pulpit/godot/scene/main/node.cpp:293)
[11] Node::_propagate_exit_tree() (/home/rafal/Pulpit/godot/scene/main/node.cpp:279 (discriminator 2))
[12] Node::_propagate_exit_tree() (/home/rafal/Pulpit/godot/scene/main/node.cpp:279 (discriminator 2))
[13] Node::_propagate_exit_tree() (/home/rafal/Pulpit/godot/scene/main/node.cpp:279 (discriminator 2))
[14] Node::_set_tree(SceneTree*) (/home/rafal/Pulpit/godot/scene/main/node.cpp:2536)
[15] Node::remove_child(Node*) (/home/rafal/Pulpit/godot/scene/main/node.cpp:1264 (discriminator 2))
[16] Node::_notification(int) (/home/rafal/Pulpit/godot/scene/main/node.cpp:158)
[17] Node::_notificationv(int, bool) (/home/rafal/Pulpit/godot/./scene/main/node.h:46 (discriminator 14))
[18] CanvasItem::_notificationv(int, bool) (/home/rafal/Pulpit/godot/./scene/2d/canvas_item.h:166)
[19] Control::_notificationv(int, bool) (/home/rafal/Pulpit/godot/./scene/gui/control.h:51)
[20] Container::_notificationv(int, bool) (/home/rafal/Pulpit/godot/./scene/gui/container.h:38)
[21] BoxContainer::_notificationv(int, bool) (/home/rafal/Pulpit/godot/./scene/gui/box_container.h:38)
[22] VBoxContainer::_notificationv(int, bool) (/home/rafal/Pulpit/godot/./scene/gui/box_container.h:81)
[23] Object::notification(int, bool) (/home/rafal/Pulpit/godot/core/object.cpp:957)
[24] Object::_predelete() (/home/rafal/Pulpit/godot/core/object.cpp:387)
[25] predelete_handler(Object*) (/home/rafal/Pulpit/godot/core/object.cpp:2000)
[26] void memdelete<Node>(Node*) (/home/rafal/Pulpit/godot/./core/os/memory.h:117)
[27] EditorInspector::_clear() (/home/rafal/Pulpit/godot/editor/editor_inspector.cpp:1765)
[28] EditorInspector::update_tree() (/home/rafal/Pulpit/godot/editor/editor_inspector.cpp:1394)
[29] EditorInspector::_notification(int) (/home/rafal/Pulpit/godot/editor/editor_inspector.cpp:2167)
[30] EditorInspector::_notificationv(int, bool) (/home/rafal/Pulpit/godot/editor/editor_inspector.h:251 (discriminator 14))
[31] Object::notification(int, bool) (/home/rafal/Pulpit/godot/core/object.cpp:957)
[32] SceneTree::_notify_group_pause(StringName const&, int) (/home/rafal/Pulpit/godot/scene/main/scene_tree.cpp:969)
[33] SceneTree::idle(float) (/home/rafal/Pulpit/godot/scene/main/scene_tree.cpp:518 (discriminator 3))
[34] Main::iteration() (/home/rafal/Pulpit/godot/main/main.cpp:1877)
[35] OS_X11::run() (/home/rafal/Pulpit/godot/platform/x11/os_x11.cpp:2975)
[36] /usr/bin/godot(main+0xdc) [0x1156893] (/home/rafal/Pulpit/godot/platform/x11/godot_x11.cpp:56)
[37] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xe7) [0x7f4b9985bb97] (??:0)
[38] /usr/bin/godot(_start+0x2a) [0x11566fa] (??:?)
```

**Comment by QbieShay** [comment]
@hpvb have fun

**Comment by akien-mga** [comment]
Doesn't crash for me on Linux with a current master build (36de6c1e3c). Can you try with 3.1 beta 8 or later?

**Comment by alimalkhalifa** [comment]
@akien-mga still crashes for me on 3.1 beta 9

**Comment by Chaosus** [comment]
Meh, tool always been dangerous territory..

**Comment by akien-mga** [comment]
I can't reproduce it in the current master branch (0d72812).

Judging by the backtrace, this might well have been fixed by b376273f1b30b1187731b318c0f6eb0447acd289.

Please comment if you can still reproduce it in 3.1 RC1 or RC2 when it's published.
